tests/unit/test_lsr_role2collection.py

```diff
@@ -0,0 +1,413 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+"""unit tests for lsr_role2collection"""
+
+import os
+import re
+import shutil
+import tempfile
+from pathlib import Path
+import unittest
+
+from lsr_role2collection import (
+    file_replace,
+    copy_tree_with_replace,
+    cleanup_symlinks,
+    import_replace,
+    from_replace,
+    gather_module_utils_parts,
+    add_rolename,
+)
+
+src_path = os.environ.get("COLLECTION_SRC_PATH", "/var/tmp/linux-system-roles")
+dest_path = os.environ.get("COLLECTION_DEST_PATH", "/var/tmp/collections")
+namespace = os.environ.get("COLLECTION_NAMESPACE", "fedora")
+collection_name = os.environ.get("COLLECTION_NAME", "system_roles")
+rolename = "systemrole"
+
+test_yaml_str = "# SPDX-License-Identifier: MIT\n---\n- name: Ensure that the role runs with default parameters\n  hosts: all\n  tasks:\n    - name: default task\n      {0}:\n        {1} {2}{3}{4}\n"
```

Comment:
You should use a multiline string with `textwrap.dedent` as in https://github.com/linux-system-roles/auto-maintenance/pull/13/files#diff-30cbdea1a4bfc877c6034e8e28d83700R478